examples/ct/handlers_test.go

```diff
@@ -864,43 +846,89 @@ func TestGetEntriesRejectsMissingParams(t *testing.T) {
 }
 
 func TestGetEntriesRanges(t *testing.T) {
+	var tests = []struct {
+		start int64
+		end   int64
+		want  int
+		desc  string
+		rpc   bool
+	}{
+		{-1, 0, http.StatusBadRequest, "-ve start value not allowed", false},
+		{0, -1, http.StatusBadRequest, "-ve end value not allowed", false},
+		{20, 10, http.StatusBadRequest, "invalid range end>start", false},
+		{3000, -50, http.StatusBadRequest, "invalid range, -ve end", false},
+		{10, 20, http.StatusInternalServerError, "valid range", true},
+		{10, 10, http.StatusInternalServerError, "valid range, one entry", true},
+		{10, 9, http.StatusBadRequest, "invalid range, edge case", false},
+		{1000, 50000, http.StatusBadRequest, "range too large to be accepted", false},
+	}
+
 	// This tests that only valid ranges make it to the backend for get-entries.
 	// We're testing request handling up to the point where we make the RPC so arrange for
 	// it to fail with a specific error.
-	for _, testCase := range getEntriesRangeTestCases {
+	for _, test := range tests {
 		mockCtrl := gomock.NewController(t)
-
 		client := mockclient.NewMockTrillianLogClient(mockCtrl)
 
-		if testCase.rpcExpected {
-			client.EXPECT().GetLeavesByIndex(deadlineMatcher(), &trillian.GetLeavesByIndexRequest{LeafIndex: buildIndicesForRange(testCase.start, testCase.end)}).Return(nil, errors.New("RPCMADE"))
+		if test.rpc {
+			client.EXPECT().GetLeavesByIndex(deadlineMatcher(), &trillian.GetLeavesByIndexRequest{LeafIndex: buildIndicesForRange(test.start, test.end)}).Return(nil, errors.New("RPCMADE"))
 		}
 
 		c := LogContext{rpcClient: client, timeSource: fakeTimeSource, rpcDeadline: time.Millisecond * 500}
 		handler := appHandler{context: c, handler: getEntries, method: http.MethodGet}
-
-		path := fmt.Sprintf("/ct/v1/get-entries?start=%d&end=%d", testCase.start, testCase.end)
+		path := fmt.Sprintf("/ct/v1/get-entries?start=%d&end=%d", test.start, test.end)
 		req, err := http.NewRequest("GET", path, nil)
-
 		if err != nil {
 			t.Fatal(err)
 		}
-
 		w := httptest.NewRecorder()
 		handler.ServeHTTP(w, req)
 
-		if expected, got := testCase.expectedStatus, w.Code; expected != got {
-			t.Fatalf("expected status %d, got %d for test case %s", expected, got, testCase.explanation)
+		if got := w.Code; got != test.want {
+			t.Errorf("GetEntries(%d, %d)=%d; want %d for test  %s", test.start, test.end, got, test.want, test.desc)
+		} else if test.rpc && !strings.Contains(w.Body.String(), "RPCMADE") {
```

Comment:
I've split the tests but not done a `continue` (which would jump over the `mockCtrl.Finish()`.